.github/workflows/update-ims.js

```diff
@@ -0,0 +1,114 @@
+const https = require('https');
+const { execSync } = require('child_process');
+const fs = require('fs');
+
+const fetchImsScript = () =>
+  new Promise((resolve, reject) => {
+    https
+      .get('https://auth.services.adobe.com/imslib/imslib.min.js', (res) => {
+        if (res.statusCode < 200 || res.statusCode >= 300) {
+          return reject(new Error('statusCode=' + res.statusCode));
+        }
+
+        let data = '';
+
+        res.on('data', (chunk) => {
+          data += chunk;
+        });
+
+        res.on('end', () => {
+          resolve({
+            data,
+            headers: res.headers,
+          });
+        });
+      })
+      .on('error', (err) => {
+        reject(err);
+      });
+  });
+
+const main = async ({ github, context, local = false }) => {
+  const response = await fetchImsScript();
+  const lastModified = new Date(response.headers['last-modified']);
+  const now = new Date();
+  const diffHours = Math.ceil((now - lastModified) / 1000 / 60 / 60);
+  console.log(
+    `imslib.min.js last modified: ${lastModified}. Last modified ${diffHours} hours ago <= 24H, means new PR created.`
+  );
+
+  if (diffHours <= 24 || local) {
+    console.log('Diff detected in the last 24h, creating PR.');
+    const header = `// Last modified ${lastModified.toISOString()}\n// Build at ${new Date().toISOString()}\n`;
+
+    if (local) return console.log('Local mode - skipped git commands.');
+
+    // If you run this on local, this might overwrite ur current git configs.
+    execSync('git config --global user.name "GitHub Action"');
+    execSync('git config --global user.email "action@github.com"');
+    execSync('git fetch');
+    execSync('git checkout stage');
+    try {
+      execSync('git branch -D update-imslib');
+    } catch (error) {
+      console.log('update-imslib branch not found, skipping delete');
+    }
+    execSync('git checkout -b update-imslib');
+    fs.writeFileSync(
+      `${local ? '../../' : ''}libs/deps/imslib.min.js`,
+      header + response.data
+    );
+    execSync('git add libs/deps/imslib.min.js');
+    execSync('git commit -m "Update imslib.min.js"');
+    execSync('git push --force origin update-imslib');
+
+    await github.rest.pulls.create({
+      owner: context.repo.owner,
+      repo: context.repo.repo,
+      title: 'Update imslib.min.js',
+      head: 'update-imslib',
+      base: 'stage',
```

Comment:
Should we consider adding any labels and/or reviewers?

Reply:
I'm on this one currently, it definitely should have labels & reviewers!